app/services/encrypted_attribute.rb

```diff
@@ -0,0 +1,76 @@
+class EncryptedAttribute
+  attr_accessor :user_access_key
+  attr_reader :encrypted, :decrypted
+
+  def self.new_user_access_key(cost: nil, key: nil)
+    env = Figaro.env
+    key ||= env.attribute_encryption_key
+    cost ||= env.attribute_cost
+    UserAccessKey.new(password: key, salt: key, cost: cost)
+  end
+
+  def self.new_from_decrypted(decrypted, user_access_key = new_user_access_key)
+    encryptor = Pii::PasswordEncryptor.new
+    encrypted = encryptor.encrypt(decrypted, user_access_key)
+    new(encrypted, decrypted: decrypted, user_access_key: user_access_key)
+  end
+
+  def initialize(encrypted, decrypted: nil, cost: nil, user_access_key: nil)
```

Comment:
Yes I am trying to train myself to use named args.

I don't think we have a method ordering convention. My personal one is class methods, then `initialize`, then instance methods. The `initialize` method is an instance method, strictly speaking, but is used to bridge the class/instance barrier, so I think of it as the bridge method in my order, a signpost that says to me "hey everything after this is instance methods".